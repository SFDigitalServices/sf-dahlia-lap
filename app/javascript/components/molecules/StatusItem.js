import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import StatusPill from '../atoms/StatusPill'

import StatusItemShape from '../../utils/shapes/StatusItemShape'
import { getSubStatusLabel } from '../../utils/statusUtils'

const StatusDate = ({ timestamp }) => (
  <div className='status-item-date'>
    {moment.unix(timestamp).format('MMM D, YYYY')}
  </div>
)

StatusDate.propTypes = { timestamp: PropTypes.number }

const StatusItem = ({ statusItem }) => {
  const { status, substatus, comment, timestamp } = statusItem

  const substatusLabel = getSubStatusLabel(status, substatus)

  return (
    <div className='status-item'>
      <div className='status-item-header'>
        <StatusPill status={status} />
        <StatusDate timestamp={timestamp} />
      </div>
      {substatusLabel && (
        <div className='item-substatus-text'>
          {substatusLabel}
        </div>
      )}
      <div className='status-item-text'>
        {comment}
      </div>
    </div>
  )
}

StatusItem.propTypes = {
  statusItem: PropTypes.shape(StatusItemShape).isRequired
}

export default StatusItem
