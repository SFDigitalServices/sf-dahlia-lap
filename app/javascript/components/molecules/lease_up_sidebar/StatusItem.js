import React from 'react'

import moment from 'moment'
import PropTypes from 'prop-types'

import StatusItemShape from '../../../utils/shapes/StatusItemShape'
import { getSubStatusLabel } from '../../../utils/statusUtils'
import StatusPill from '../../atoms/StatusPill'

const StatusDate = ({ timestamp }) => (
  <div className='status-item-date'>{moment.unix(timestamp).format('MMM D, YYYY')}</div>
)
const StatusCreatedBy = ({ createdBy }) => <div className='status-item-created-by'>{createdBy}</div>

StatusDate.propTypes = { timestamp: PropTypes.number }

const StatusItem = ({ statusItem }) => {
  const { status, substatus, comment, timestamp, created_by } = statusItem

  const substatusLabel = getSubStatusLabel(status, substatus)

  return (
    <div className='status-item'>
      <div className='status-item-header'>
        <StatusPill status={status} />
      </div>
      {substatusLabel && <div className='item-substatus-text'>{substatusLabel}</div>}
      <div className='status-item-text'>{comment}</div>
      <div className='status-item-footer'>
        <StatusCreatedBy createdBy={created_by} />
        <StatusDate timestamp={timestamp} />
      </div>
    </div>
  )
}

StatusItem.propTypes = {
  statusItem: PropTypes.shape(StatusItemShape).isRequired
}

export default StatusItem
