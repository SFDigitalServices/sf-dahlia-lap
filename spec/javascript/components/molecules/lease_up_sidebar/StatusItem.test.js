import React from 'react'

import { mount, shallow } from 'enzyme'

import StatusItem from 'components/molecules/lease_up_sidebar/StatusItem'

import { mockStatusItem } from '../../../mocks/statusItemMock'

const getWrapper = (item) => shallow(<StatusItem statusItem={item} />)

describe('StatusItem', () => {
  test('should render an approved status with substatus and comments correctly', () => {
    const wrapper = getWrapper(mockStatusItem())
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly when no substatus provided', () => {
    const wrapper = getWrapper(mockStatusItem({ substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly when no comment provided', () => {
    const wrapper = getWrapper(mockStatusItem({ comment: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly when no comment or substatus provided', () => {
    const wrapper = getWrapper(mockStatusItem({ substatus: null, comment: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with withdrawn status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Withdrawn', substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Processing status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Processing', substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Appealed status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Appealed', substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Disqualified status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Disqualified', substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Lease Signed status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Lease Signed', substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should format timestamps as expected', () => {
    const wrapper = mount(
      <StatusItem statusItem={mockStatusItem({ status: 'Lease Signed', substatus: null })} />
    )
    const timestamp = wrapper.find('StatusDate').text()
    expect(timestamp).toEqual('Aug 25, 2020')
  })
})
