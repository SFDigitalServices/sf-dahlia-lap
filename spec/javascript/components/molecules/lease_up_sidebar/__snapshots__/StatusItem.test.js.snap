// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusItem should render an approved status with substatus and comments correctly 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-approved"
      >
        Approved
      </div>
    </div>
    <div
      class="item-substatus-text"
    >
      Approval letter sent
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly when no comment or substatus provided 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-approved"
      >
        Approved
      </div>
    </div>
    <div
      class="status-item-text"
    />
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly when no comment provided 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-approved"
      >
        Approved
      </div>
    </div>
    <div
      class="item-substatus-text"
    >
      Approval letter sent
    </div>
    <div
      class="status-item-text"
    />
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly when no substatus provided 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-approved"
      >
        Approved
      </div>
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly with Appealed status 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-appealed"
      >
        Appealed
      </div>
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly with Disqualified status 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-disqualified"
      >
        Disqualified
      </div>
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly with Lease Signed status 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-leased"
      >
        Lease Signed
      </div>
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly with Processing status 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-processing"
      >
        Processing
      </div>
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusItem should render properly with withdrawn status 1`] = `
<DocumentFragment>
  <div
    class="status-item"
    data-testid="status-item"
  >
    <div
      class="status-item-header"
    >
      <div
        class="status-pill is-withdrawn"
      >
        Withdrawn
      </div>
    </div>
    <div
      class="status-item-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
    </div>
    <div
      class="status-item-footer"
    >
      <div
        class="status-item-created-by"
      />
      <div
        class="status-item-date"
      >
        Aug 25, 2020
      </div>
    </div>
  </div>
</DocumentFragment>
`;
